frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
+                          const activeColumnBData =
+                              activeColumn === ColumnDataEnum.Date
+                                  ? Date.parse(
+                                        b[
+                                            activeColumn as keyof typeof ColumnDataEnum
+                                        ]
+                                    )
+                                  : b[
+                                        activeColumn as keyof typeof ColumnDataEnum
+                                    ];
+
+                          if (activeColumnSortOrder === FilterOptionsEnum.ASC) {
+                              return activeColumnAData > activeColumnBData
+                                  ? 1
+                                  : -1;
+                          } else {
+                              return activeColumnAData < activeColumnBData
+                                  ? 1
+                                  : -1;
+                          }
+                      }
+                  )
+                : normalizedData,
+        [activeColumn, activeColumnSortOrder, normalizedData]
+    );
     return (
         <>
             <h2>Receivers:</h2>
-            {receiverDetails?.map((receiver, i) => (
-                <div
-                    className="display-flex flex-column margin-bottom-5"
-                    key={"receiver_" + i}
+            <div className="message-receivers-table">
+                <Table
+                    key="messagedetails"
+                    aria-label="Message Details"
+                    bordered
                 >
-                    <DetailItem
-                        item={"Receiver Name"}
-                        content={receiver.receivingOrg}
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Receiver Service"}
-                        content={receiver.receivingOrgSvc}
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Incoming Report Id"}
-                        content={receiver.reportId}
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Incoming File Name"}
-                        content={receiver.fileName}
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Incoming File URL"}
-                        content={receiver.fileUrl}
-                    ></DetailItem>
-                    <DetailItem
-                        item={"Date/Time Submitted"}
-                        content={formattedDateFromTimestamp(
-                            receiver.createdAt,
-                            "MMMM DD YYYY"
+                    <thead>
+                        <tr>
+                            <MessageReceiversCol
+                                columnHeaderTitle={ColumnDataEnum.Name}
+                                activeColumn={activeColumn}
```

Comment:
For common props like this, we could just store them in a spread variable:

```tsx
const commonProps = {
    activeColumn,
    setActiveColumn,
    activeColumnSortOrder,
    setActiveColumnSortOrder
};

<MessageReceiversCol
    { ...commonProps }
    columnHeaderTitle={ColumnDataEnum.Name}
    rowSpan={2}
/>
<MessageReceiversCol
    { ...commonProps }
    columnHeaderTitle={ColumnDataEnum.Service}
    rowSpan={2}
/>
...
```

Reply:
I like this! Will change.